src/components/categories/Edit.vue

```diff
@@ -0,0 +1,216 @@
+	id: number
+}>()
+
+const updateError = ref('')
+
+const token = useSessionToken()
+
+const router = useRouter()
+
+const categories = new Categories({
+	baseUrl: import.meta.env.VITE_BACKEND_URL
+})
+
+const warnBeforeLeave = ref(true)
+
+const updateRequest = ref<UpdateCategoryRequest>({
+	name: '',
+	slug: ''
+})
+
+const {
+	state: cat,
+	error,
+	isLoading,
+	execute
+} = useAsyncState(async () => {
+	const resp = await categories.getCategory(props.id)
+	updateRequest.value.info = resp.data.info || ''
+	updateRequest.value.name = resp.data.name
+	updateRequest.value.slug = resp.data.slug
+	updateRequest.value.sortDirection = resp.data.sortDirection
+	return resp.data
+}, null)
+
+onBeforeRouteLeave(() => {
+	if (
+		warnBeforeLeave.value &&
+		(cat.value?.name !== updateRequest.value.name ||
+			cat.value?.slug !== updateRequest.value.slug ||
+			cat.value?.info !== updateRequest.value.info ||
+			cat.value?.sortDirection !== updateRequest.value.sortDirection)
+	) {
+		if (!window.confirm('Do you want to leave? You have unsaved changes.')) {
+			return false
+		}
+	}
+})
+
+const errorResponse = computed(
+	() => error.value as HttpResponse<unknown, void | ProblemDetails>
+)
+
+async function submit() {
+	useApi(
+		() =>
+			categories.updateCategory(
+				props.id,
+				{
+					name: updateRequest.value.name,
+					info: updateRequest.value.info,
+					slug: updateRequest.value.slug,
+					sortDirection: updateRequest.value.sortDirection
+				},
+				useAuth(token.value)
+			),
+		() => {
+			warnBeforeLeave.value = false
+			router.push({ name: 'categoryView', params: { id: props.id } })
+		},
+		(error) => {
+			updateError.value = 'Failed to update: ' + error.status.toString(10)
+		}
+	)
+}
+</script>
+
+<template>
+	<div class="container">
+		<div v-if="isLoading">Loading...</div>
+		<div v-else-if="error" class="error-container">
+			<p class="errorText">
+				<!-- For unexpected server errors; fields will be empty, hence the defaults. -->
+				Failed to fetch category: {{ errorResponse.status }}
+				{{ errorResponse.error?.title ?? 'Check the console for more info.' }}
+			</p>
+			<button @click="execute()" class="button">Reload</button>
+		</div>
+
+		<div v-else class="main-content">
+			<h1 class="title">Update {{ cat?.name }}</h1>
+
+			<RouterLink
+				class="back-link"
+				:to="{ name: 'categoryView', params: { id } }"
+				title="Changes will not be saved."
+				>&lt; Back</RouterLink
+			>
+
+			<p v-if="updateError" class="error-text">{{ updateError }}</p>
+
+			<form @submit.prevent="submit">
+				<table>
+					<tbody>
+						<tr>
+							<th>ID:</th>
+							<td>{{ cat?.id }}</td>
+						</tr>
+						<tr>
+							<th>
+								<label for="name">Name:</label>
+							</th>
+							<td>
+								<input v-model="updateRequest.name" id="name" />
+							</td>
+						</tr>
+						<tr>
+							<th>
+								<label for="slug">Slug:</label>
+							</th>
+							<td>
+								<input v-model="updateRequest.slug" id="slug" />
```

Comment:
Put the same validation on this and the other controls as we have on the create page. You could maybe even use the same component for both.

Reply:
I've noticed that we don't have the same thing in the LB Edit Page either. I'll go ahead and create a component to be used in all four pages/